blackbox/sqllogictest/src/sqllogictest.py

```diff
@@ -297,6 +300,7 @@ def run_file(fh, hosts, log_level, log_file, failfast):
                 else:
                     logger.debug('%s; %s', cmd[1], 'Query is whitelisted', extra=attr)
     finally:
+        fh.close()
```

Comment:
I think we should change run_file to take a filename and do the `open` inside `run_file` as well to keep open/close together.